notebooks/224-stable-diffusion-text-to-image/224-stable_dffusion-text-to-image.ipynb

```diff
@@ -0,0 +1,1271 @@
+        "\n",
+        "During latent diffusion training, the encoder is used to get the latent representations (latents) of the images for the forward diffusion process, which applies more and more noise at each step. During inference, the denoised latents generated by the reverse diffusion process are converted back into images using the VAE decoder. As we will see during inference we **only need the VAE decoder**.\n",
+        "\n",
+        "In our inference pipeline, we will need only decoding part of VAE, but forward function, which used for tracing runs encoding too. For obtaining only necessary part of model, we will wrap it into new model class."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "fQvd38qPHLtq"
+      },
+      "outputs": [],
+      "source": [
+        "class VAEDecoderWrapper(torch.nn.Module):\n",
+        "  def __init__(self, vae):\n",
+        "      super().__init__()\n",
+        "      self.vae = vae\n",
+        "\n",
+        "  def forward(self, latents):\n",
+        "      return self.vae.decode(latents)\n",
+        "\n",
+        "vae_decoder = VAEDecoderWrapper(pipe.vae)\n",
+        "\n",
+        "vae_decoder.eval()\n",
+        "with torch.no_grad():\n",
+        "    torch.onnx.export(vae_decoder, latents, 'vae_decoder.onnx', input_names=['latents'], output_names=['sample'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Convert models using OpenVINO Model Optimizer API"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "While ONNX models are directly supported by OpenVINO™, it can be useful to convert them to IR format to take advantage of advanced OpenVINO optimization tools and features. `mo.convert' function can be used for converting model using OpenVINO Model Optimizer capabilities. It returns of instance OpenVINO Model class, which is ready to use in python interface and can be serialized to IR for next usages."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 20,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "from openvino.tools import mo\n",
+        "from openvino.runtime import serialize\n",
+        "\n",
+        "text_encoder = mo.convert(input_model='text-encoder.onnx', data_type='FP16')\n",
+        "serialize(text_encoder, 'text_encoder.xml')\n",
+        "unet = mo.convert(input_model='unet/unet.onnx', data_type='FP16')\n",
+        "serialize(unet, 'unet.xml')\n",
+        "vae_decoder = mo.convert(input_model='vae_decoder.onnx', data_type='FP16')\n",
+        "serialize(vae_decoder, 'vae_decoder.xml')"
```

Comment:
`mo.convert` will likely be called differently in the final proposal. As well likely as `data_type='FP16'` won't be required. Are you going to merge soon or keep until we change it?

Reply:
I do not know, probably this notebook can not wait release (otherwise it will be non so actual like cristams tree in the march :)), so I prefer to switch back to mo script usage for now and then to upgrade notebook.

There is also some memory limitation (models too large to store them all in RAM, probably one step conversion without saving onnx on disk, it will be hard to recover on machines with small RAM, do not know should it be considered for mo convert or not, just my note:))